blackbox_repository/tabulated_benchmark.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from pathlib import Path
+from typing import List, Optional
+
+from blackbox_repository.blackbox_tabular import BlackboxTabular
+from syne_tune.backend.simulator_backend.simulator_backend import SimulatorBackend
+from syne_tune.backend.trial_status import Status
+
+
+class TabulatedBenchmark(SimulatorBackend):
+
+    def __init__(
+            self,
+            blackbox: BlackboxTabular,
```

Comment:
Tabular is employed for datasets that are complete (both NB201 and fcnet are), surrogate is employed for datasets that are incomplete (just a sample of evaluations but not the whole space).

Reply:
But surrogate makes just as much sense on a tabular (complete) dataset, maybe even more so, because interpolation is easier.

This is really important. If I run bayesopt on fcnet, I must not use this search space with combinatorial parameters, this would work really poorly.

Maybe it would be best to call this class here BlackboxSimulatorBackend, and allow for blackbox: Blackbox, then the user could specify either BlackboxTabular or BlackboxSurrogate.

The thing is: If you run bayesopt on a search space with lots of categoricals, you get very poor results, we don't wamnt that.

Reply:
Note that this comment is made on outdated code. `BlackboxTabular` was not used anymore as the argument of this function. I just a change so that it is not explicitly `Blackbox`.

> But surrogate makes just as much sense on a tabular (complete) dataset, maybe even more so, because interpolation is easier.

Interpolation is also possible on fcnet, surrogate can be used in this case as well with the current code.